### PR TITLE
Connect form to firebase

### LINKED ISSUE #18: Connect form to firebase

### PR DESCRIPTION
**Pull request by biggicode**

### CODE REVIEW

**Review by MariaTeo — APPROVED** [review]

File: src/component/auction-page/auction-page.component.jsx [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
+import { useSelector } from "react-redux";
+import { useLocation, useParams } from "@reach/router";
+import { db } from "../../config/firebase";
 import * as S from "./auction-page.style";
 import image from "../../asset/images/phone.jpg";
 
-export const AuctionPage = () => {
+export const AuctionPage = ({ price, description }) => {
+  const isLoggedIn = useSelector(({ user }) => user.isLoggedIn);
+  const location = useLocation();
```
Comment (marianzburlea) [inline src/component/auction-page/auction-page.component.jsx]: you are no longer using location

File: src/component/auction-page/auction-page.component.jsx [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
+  const { id } = useParams();
+  const { title, StartingPrice } = location.state;
```
Comment (marianzburlea) [inline src/component/auction-page/auction-page.component.jsx]: `title` and `StartingPrice` are not used in this file

File: src/component/auction-page/auction-page.component.jsx [hunk 3]
```diff
@@ -1,31 +1,38 @@
...
+  const [auction, setAuction] = useState({});
+
+  useEffect(() => {
+    db.collection("auctions")
+      .doc(id)
+      .get()
+      .then((snapshot) => setAuction(snapshot.data()));
+  }, []);
+
+  console.log(auction);
```
Comment (marianzburlea) [inline src/component/auction-page/auction-page.component.jsx]: remove console.log

File: src/component/auctions/auctions.component.jsx [hunk 4]
```diff
@@ -16,10 +34,9 @@ export const Auctions = () => {
           <SA.StyledOrderBy>Start Price - high to low</SA.StyledOrderBy>
           <SA.StyledOrderBy>Start Price - low to high</SA.StyledOrderBy>
         </SA.StyledFilter>
-        <ActionCard />
-        <ActionCard />
-        <ActionCard />
-        <ActionCard />
+        {auctionsList.map(({ title, startingPrice, id }) => {
+          return <ActionCard key={id} id={id} />;
+        })}
```
Comment (marianzburlea) [inline src/component/auctions/auctions.component.jsx]: you can have implicit return

**Review by marianzburlea — CHANGES_REQUESTED** [review]
small updates :)
please clean up the code

File: src/component/action-card/action-card.component.jsx [hunk 5]
```diff
@@ -2,18 +2,18 @@ import React from 'react'
 import * as ACS from './action-card.style'
 import Image from '../../asset/images/bicycle.jpg'
 
-export const ActionCard = () => {
+export const ActionCard = ({id}) => {
```
Comment (marianzburlea) [inline src/component/action-card/action-card.component.jsx]: please use the formatter, `CTRL + SHIFT + F`, it's not that difficult.
also make a note somewhere so I don't have to remind you so many times.

File: src/component/action-card/action-card.component.jsx [hunk 6]
```diff
@@ -2,18 +2,18 @@ import React from 'react'
...
 
   return(
     <ACS.ActionCard>
       <ACS.ActionCardHeader>20:35:21</ACS.ActionCardHeader>
       <ACS.ActionCardBody>
         <ACS.ActionCardImage src={Image} />
-        <ACS.ProductName>A big white bicycle with 2 rims</ACS.ProductName>
+        {/* <ACS.ProductName>{title}</ACS.ProductName> */}
```
Comment (marianzburlea) [inline src/component/action-card/action-card.component.jsx]: remove comments

File: src/component/action-card/action-card.component.jsx [hunk 7]
```diff
@@ -2,18 +2,18 @@ import React from 'react'
...
         <ACS.PriceContainer>
           <ACS.Price>
             <ACS.PriceText>Start Price</ACS.PriceText>
-            <ACS.PriceValue>$ 120.00</ACS.PriceValue>
+            {/* <ACS.PriceValue>$ {startingPrice}</ACS.PriceValue> */}
```
Comment (marianzburlea) [inline src/component/action-card/action-card.component.jsx]: remove comments

File: src/component/sell/sell.component.jsx [hunk 8]
```diff
@@ -1,11 +1,13 @@
...
+import { Formik, Field } from "formik";
 import { initialValues, validationSchema } from "./sell.form";
+import { db } from "../../config/firebase";
 
 export const Sell = () => {
-  const onSubmit = values => {
-    console.log(values)
+  const onSubmit = (values, {resetForm}) => {
+    db.collection("auctions").add(values)
+    resetForm({ values: ''})
```
Comment (marianzburlea) [inline src/component/sell/sell.component.jsx]: why do you pass an object with a key *values* and an empty string as value?

File: src/component/sell/sell.form.js [hunk 9]
```diff
@@ -1,24 +1,24 @@
 import * as yup from 'yup'
 
-export const initialValues = {firstName: '', lastName: '', description: '', startingPrice: '', currency: '', city: '', email: '', phone: ''}
+export const initialValues = {title: '', images: '', description: '', startingPrice: '', currency: '', city: '', email: '', phone: ''}
```
Comment (marianzburlea) [inline src/component/sell/sell.form.js]: use the formatter again.

**Review by marianzburlea — CHANGES_REQUESTED** [review]
further cleaning of the code is required

File: src/component/auction-card/auction-card.component.jsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from "react";
+import * as ACS from "./auction-card.style";
+import Image from "../../asset/images/bicycle.jpg";
+import { db } from "../../config/firebase";
+
+export const AuctionCard = ({ id }) => {
+  const [CardData, setCardData] = useState({});
```
Comment (marianzburlea) [inline src/component/auction-card/auction-card.component.jsx]: `CardData` => `cardData`

File: src/component/auctions/auctions.component.jsx [hunk 11]
```diff
@@ -1,9 +1,27 @@
...
+      let list = [];
+
+      snap.forEach((doc) => {
+        list.push({
+          id: doc.id,
+          ...doc.data(),
+        });
+      });
+      setAuctionsList(list);
+    });
```
Comment (marianzburlea) [inline src/component/auctions/auctions.component.jsx:20]: unsubscribe from the stream as well.

File: src/component/home/home.component.jsx [hunk 12]
```diff
@@ -1,112 +1,115 @@
...
-    []
-  )
+      snap.forEach((doc) => {
+        list.push({
+          id: doc.id,
+          ...doc.data(),
+        });
+      });
+      setNameList(list);
+    });
```
Comment (marianzburlea) [inline src/component/home/home.component.jsx:22]: unsubscribe in useEffect

File: src/component/home/home.component.jsx [hunk 13]
```diff
@@ -1,112 +1,115 @@
...
-          <div>{anaaremere ? "da" : "nu"}</div>
-        </>
-      ))}
+      <input
+        type="text"
+        value={name}
+        onChange={handleChange}
+        onKeyPress={handlePress}
+      />
+      <div
```
Comment (marianzburlea) [inline src/component/home/home.component.jsx]: why css in js when you use styled-components?

**Review by marianzburlea — CHANGES_REQUESTED** [review]

File: src/component/auction-card/auction-card.component.jsx [hunk 14]
```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from "react";
+import * as ACS from "./auction-card.style";
+import Image from "../../asset/images/bicycle.jpg";
```
Comment (tandaraelena) [inline src/component/auction-card/auction-card.component.jsx]: Image => image

File: src/component/sell/sell.component.jsx [hunk 15]
```diff
@@ -1,42 +1,41 @@
...
 
-  
   return (
     <S.StyledSell>
-
       <Formik
         initialValues={initialValues}
         onSubmit={onSubmit}
         validationSchema={validationSchema}
+        enebleReinitialize
```
Comment (tandaraelena) [inline src/component/sell/sell.component.jsx]: Maybe a typo for 'enableReinitialize' ?

File: src/component/app/app.component.jsx [hunk 16]
```diff
@@ -1,36 +1,35 @@
...
         <Footer />
       </Provider>
     </ThemeProvider>
-  )
-}
+  );
+};
 
-export { App }
+export { App };
```
Comment (QuintyHH) [inline src/component/app/app.component.jsx]: Any reason we are exporting like this?
If you want a named export, just use the keyword 'export' when declaring the App component (line 18).
If you want a default export, use 'export default App' here.

Your current implementation exports a new Object that has App as property. No reason to declare this new Object.

File: src/component/auction-page/auction-page.component.jsx [hunk 17]
```diff
@@ -35,7 +37,11 @@ export const AuctionPage = () => {
...
         <S.WhiteContainer>
-          <S.TextIfLoggedOut><S.StyledLink>Register</S.StyledLink> or <S.StyledLink>login</S.StyledLink> to place a bid and to participate on the auction.</S.TextIfLoggedOut>
+          <S.TextIfLoggedOut>
+            <S.StyledLink>Register</S.StyledLink> or{" "}
+            <S.StyledLink>login</S.StyledLink> to place a bid and to participate
+            on the auction.
+          </S.TextIfLoggedOut>
         </S.WhiteContainer>
         {!isLoggedIn && <S.RegisterButton>Register </S.RegisterButton>}
         {!isLoggedIn && <S.LogInButton>Log In</S.LogInButton>}
```
Comment (QuintyHH) [inline src/component/auction-page/auction-page.component.jsx]: No need to check isLoggedIn twice, you can replace this with 
{ isLoggedIn $$ (
  <>
        <S.RegisterButton>Register </S.RegisterButton>
        <S.LogInButton>Log In</S.LogInButton>
  </>)}

File: src/component/auctions/auctions.component.jsx [hunk 18]
```diff
@@ -1,9 +1,27 @@
...
+  }, []);
+
+  console.log(auctionsList);
```
Comment (QuintyHH) [inline src/component/auctions/auctions.component.jsx]: Hello, console :D

File: src/component/home/home.component.jsx [hunk 19]
```diff
@@ -1,112 +1,115 @@
...
-    console.log(value)
-  }
-//Snapchot for firebase with for each
-  useEffect(
-    () => {
-      db.collection("hahahe").onSnapshot(snap => {
-        let list = []
+    const { value } = e.target;
+    setName(value);
+    console.log(value);
```
Comment (QuintyHH) [inline src/component/home/home.component.jsx]: Hello, console :D

File: src/component/home/home.component.jsx [hunk 20]
```diff
@@ -1,112 +1,115 @@
...
 
-  const handlePress = e => {
+  const handlePress = (e) => {
     if (e.charCode === 13) {
       // Vreau sa il trimit la baza de date cand apas enter
       db.collection("hahahe").add({
         name,
-        anaaremere: Math.random() - 0.5 > 0
-      })
+        anaaremere: Math.random() - 0.5 > 0,
```
Comment (QuintyHH) [inline src/component/home/home.component.jsx]: Let's leave poor Ana and her apples and lets change this to something that makes a bit more sense in this context :D

File: src/component/home/home.component.jsx [hunk 21]
```diff
@@ -1,112 +1,115 @@
...
+      });
       // Vreau sa curat formularul
-      setName("")
-      e.target.focus()
+      setName("");
+      e.target.focus();
     }
-    console.log(e, Object.keys(e))
-  }
+    console.log(e, Object.keys(e));
```
Comment (QuintyHH) [inline src/component/home/home.component.jsx]: Hello, console :D

File: src/component/home/home.component.jsx [hunk 22]
```diff
@@ -1,112 +1,115 @@
...
-          The best place to buy and sell!
-          </HOS.PosterText>
-          {!isLoggedIn && <HOS.PosterButton onClick={logheazaMa}>Register</HOS.PosterButton>}
-          {isLoggedIn && <HOS.PosterButton onClick={scoateMa}>LogOUT</HOS.PosterButton>}
-          <HOS.PosterButton onClick={smartRandom}>Add Random Data</HOS.PosterButton>
+          <HOS.PosterText>The best place to buy and sell!</HOS.PosterText>
+          {!isLoggedIn && (
+            <HOS.PosterButton onClick={logheazaMa}>Register</HOS.PosterButton>
+          )}
+          {isLoggedIn && (
```
Comment (QuintyHH) [inline src/component/home/home.component.jsx]: No need to check isLoggedIn twice, we can use a ternary here. 
{!isLoggedIn 
? <HOS.PosterButton onClick={logheazaMa}>Register</HOS.PosterButton> 
: <HOS.PosterButton onClick={scoateMa}>LogOUT</HOS.PosterButton>}
}

File: src/component/sell/sell.component.jsx [hunk 23]
```diff
@@ -45,22 +44,19 @@ export const Sell = () => {
                 </div>
```
Comment (QuintyHH) [inline src/component/sell/sell.component.jsx]: What's the point of this wrapper div?

File: src/component/sell/sell.component.jsx [hunk 24]
```diff
@@ -45,22 +44,19 @@ export const Sell = () => {
...
               )}
             </Field>
-            <Field
-              name="startingPrice"
-              id="startingPrice"
-            >
+            <Field name="startingPrice" id="startingPrice">
               {({ field, meta }) => (
                 <>
                   <S.StyledLabel>Starting Price</S.StyledLabel>
```
Comment (QuintyHH) [inline src/component/sell/sell.component.jsx:62]: You're only returning one component, no need for the Fragments

File: src/component/sell/sell.component.jsx [hunk 25]
```diff
@@ -69,26 +65,20 @@ export const Sell = () => {
...
             <S.StyledFormSeparator />
             <S.StyledSectionTitle>Contact info</S.StyledSectionTitle>
-            <Field
-              name="city"
-              id="city"
-            >
+            <Field name="city" id="city">
               {({ field, meta }) => (
                 <div>
                   <S.StyledLabel>Your city</S.StyledLabel>
```
Comment (QuintyHH) [inline src/component/sell/sell.component.jsx:84]: No need for the wrapper div

File: src/component/sign-in/sign-in.component.jsx [hunk 26]
```diff
@@ -1,80 +1,84 @@
...
   const handleSignInWithGitHub = async () => {
-    try{
-      const { user: { uid, displayName, photoURL, email } } = await auth.signInWithPopup(GitHubProvider)
-      console.log(uid, displayName, photoURL, email)
-      dispatch(setUser({uid, displayName, photoURL, email}))
+    try {
+      const {
+        user: { uid, displayName, photoURL, email },
+      } = await auth.signInWithPopup(GitHubProvider);
+      console.log(uid, displayName, photoURL, email);
```
Comment (QuintyHH) [inline src/component/sign-in/sign-in.component.jsx]: Hello, console :D

File: src/component/sign-in/sign-in.component.jsx [hunk 27]
```diff
@@ -1,80 +1,84 @@
...
+      dispatch(setUser({ uid, displayName, photoURL, email }));
       closePopUp();
     } catch (error) {
-      console.log(error,"Ceva nu merge")
+      console.log(error, "Ceva nu merge");
```
Comment (QuintyHH) [inline src/component/sign-in/sign-in.component.jsx:18]: To increase visibility in dev tools, you can actually use console.error

File: src/component/sign-in/sign-in.component.jsx [hunk 28]
```diff
@@ -1,80 +1,84 @@
...
   //Function to close the pop up for sign in
   const closePopUp = () => {
-    if(userData != null) {
-        closeSignIn();
+    if (userData != null) {
+      closeSignIn();
     }
-  }
-  
+  };
```
Comment (QuintyHH) [inline src/component/sign-in/sign-in.component.jsx]: const closePopUp => userData && closeSignIn() 

Looks cleaner

File: src/component/sign-in/sign-in.component.jsx [hunk 29]
```diff
@@ -1,80 +1,84 @@
...
 
   const handleSignOut = async () => {
     try {
-      const result = await auth.signOut()
-      dispatch(setUser(null))
-      console.log(result)
+      const result = await auth.signOut();
+      dispatch(setUser(null));
+      console.log(result);
```
Comment (QuintyHH) [inline src/component/sign-in/sign-in.component.jsx]: Hello, console! :D

File: src/component/sign-in/sign-in.style.js [hunk 30]
```diff
@@ -1,4 +1,4 @@
-import styled, { css } from 'styled-components'
+import styled, { css } from "styled-components";
 
 const BasicButton = css`
   border: none;
```
Comment (QuintyHH) [inline src/component/sign-in/sign-in.style.js:4]: Why set border radius of 6px, if you are setting the border to none?

**Review by marianzburlea — APPROVED** [review]

**Review by tandaraelena — APPROVED** [review]

**Pull request merged**